subprojects/core-api/src/main/java/org/gradle/api/artifacts/type/ArtifactTypeDefinition.java

```diff
@@ -54,6 +54,30 @@
     @Incubating
     String JVM_RESOURCES_DIRECTORY = "java-resources-directory";
 
+    /**
+     * Represents the packaged C++ public header files
+     *
+     * @since 5.3
+     */
+    @Incubating
+    String C_PLUS_PLUS_API_ZIP_TYPE = "zip";
+
+    /**
+     * Represents a directory of C++ public header files
+     *
+     * @since 5.3
+     */
+    @Incubating
+    String C_PLUS_PLUS_API_DIRECTORY = "c-plus-plus-api-directory";
```

Comment:
How is this different from a raw directory?

Reply:
It's not, really, and that's how I originally implemented it.  There is, however, a distinction between directory types for the jvm directories, and it seemed odd that we would have that distinction but not do so for c headers.  Ultimately, it probably doesn't matter - I was just following the pattern already there.